snapshot/git/gitdb/schemes.go

```diff
@@ -0,0 +1,64 @@
+package gitdb
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/scootdev/scoot/snapshot"
+)
+
+// GitDB uses several different Schemes to identify, upload and download Values.
+// schemeValue abstracts these concerns.
+type schemeValue interface {
+	ID() snapshot.ID
+
+	Kind() valueKind
+	SHA() string
+	Upload(db *DB) (snapshot.ID, error)
+	Download(db *DB) (snapshot.ID, error)
+}
+
+// parseID parses ID into a schemeValue
+func parseID(id snapshot.ID) (schemeValue, error) {
+	if id == "" {
+		return nil, fmt.Errorf("empty snapshot ID")
+	}
+	parts := strings.Split(string(id), "-")
```

Comment:
Not clear from the code what a valid snapshot.ID is - I see it's in the doc, I would suggest at some point it's added to a README.md